spiffworkflow-backend/src/spiffworkflow_backend/data_stores/crud.py

```diff
@@ -0,0 +1,27 @@
+from typing import Any
+
+
+class DataStoreCRUD:
+    @staticmethod
+    def existing_data_stores() -> list[dict[str, Any]]:
+        raise Exception("must implement")
+
+    @staticmethod
```

Comment:
if used the word data store in a very generic way, such that arbitrary database tables with arbitrary properties could be included, the only thing i can see in this interface that wouldn't fit very well is that query_data_store doesn't accept all of the querying options you might want, such as filtering by column values and sorting. that being said, perhaps you wouldn't be allowed to get that fancy via this "API." it seems that you have the same limitation with kkv, right, where you can get stuff out, but you have to get everything? you can't filter by secondary key via this "API"? Perhaps there is another way you can filter by secondary key, and a similar thing could be used for these "data model data stores" that i'm hoping we build?

Reply:
we just discussed and all agree the idea is good, going to defer and do the basic crud operations first.